libraries/standard/mqtt/src/mqtt.c

```diff
@@ -1558,25 +1561,39 @@ static MQTTStatus_t receiveConnack( const MQTTContext_t * pContext,
 
 /*-----------------------------------------------------------*/
 
-static MQTTStatus_t resendPendingAcks( MQTTContext_t * pContext )
+static MQTTStatus_t handleSessionResumption( MQTTContext_t * pContext,
+                                             bool sessionPresent )
 {
     MQTTStatus_t status = MQTTSuccess;
-    MQTTStateCursor_t cursor = MQTT_STATE_CURSOR_INITIALIZER;
-    uint16_t packetId = MQTT_PACKET_ID_INVALID;
-    MQTTPublishState_t state = MQTTStateNull;
 
     assert( pContext != NULL );
 
-    /* Get the next packet Id for which a PUBREL need to be resent. */
-    packetId = MQTT_PubrelToResend( pContext, &cursor, &state );
-
-    /* Resend all the PUBREL acks after session is reestablished. */
-    while( ( packetId != MQTT_PACKET_ID_INVALID ) &&
-           ( status == MQTTSuccess ) )
+    if( sessionPresent == true )
     {
-        status = sendPublishAcks( pContext, packetId, state );
+        MQTTStateCursor_t cursor = MQTT_STATE_CURSOR_INITIALIZER;
+        uint16_t packetId = MQTT_PACKET_ID_INVALID;
+        MQTTPublishState_t state = MQTTStateNull;
 
+        /* Get the next packet ID for which a PUBREL need to be resent. */
         packetId = MQTT_PubrelToResend( pContext, &cursor, &state );
+
+        /* Resend all the PUBREL acks after session is reestablished. */
+        while( ( packetId != MQTT_PACKET_ID_INVALID ) &&
+               ( status == MQTTSuccess ) )
+        {
+            status = sendPublishAcks( pContext, packetId, state );
+
+            packetId = MQTT_PubrelToResend( pContext, &cursor, &state );
+        }
+    }
+    else
+    {
+        ( void ) memset( pContext->outgoingPublishRecords,
```

Comment:
A comment stating that publish records are being cleared due to clean session would be good to have

Reply:
Thanks, I forgot to add one here